// This proto describes the layout of column headers in the Library.
// It is used in place of QHeaderView's built-in save and restore state functions
// because those break across library schema changes.

syntax = "proto2";

package mixxx.library;

option optimize_for = LITE_RUNTIME;

message HeaderViewState {
  message HeaderState {
    optional bool hidden = 1;
    optional int32 size = 2;
    optional int32 logical_index = 3;
    optional int32 visual_index = 4;
    optional string column_name = 5;
  }
  repeated HeaderState header_state = 1;
  optional bool sort_indicator_shown = 2;
  // This is a logical index value.
  optional int32 sort_indicator_section = 3;
  // This corresponds to the Qt::SortOrder enum.
  optional int32 sort_order = 4;
}
